ultravox/data/datasets.py

```diff
@@ -1093,8 +1138,12 @@ def __init__(
             stop_strategy: Strategy for stopping iteration.
             seed: Optional seed for reproducibility.
             static: If true, the datasets are interleaved in a static order with equal weights.
+
+        Note: This class inherits from SizedIterableDataset for type compatibility,
+        but implements its functionality independently. It does not call
+        super().__init__() as it operates on multiple datasets rather than a single one,
+        and it doesn't rely on the parent's implementation.
         """
-        super().__init__()
```

Comment:
Failing to call `__init__` is generally not a good idea and seems to be a bad smell to me.

In this particular case it seems you can simply call `len(self)` once `self._datasets` is set to get num_samples.

`self._dataset` however is never set because of this and that means this class isn't a sub-class of `SizedIterableDataset`.